django/thunderstore/core/settings.py

```diff
@@ -487,10 +497,23 @@ def show_debug_toolbar(request):
 AWS_S3_OBJECT_PARAMETERS = {
     "CacheControl": "max-age=2592000",  # 30 days
 }
+AWS_S3_SECURE_URLS = env.bool("AWS_S3_SECURE_URLS")
+
+# Usermedia S3 settings
+
+USERMEDIA_S3_ENDPOINT_URL = env.str("USERMEDIA_S3_ENDPOINT_URL")
+USERMEDIA_S3_ACCESS_KEY_ID = env.str("USERMEDIA_S3_ACCESS_KEY_ID")
+USERMEDIA_S3_SECRET_ACCESS_KEY = env.str("USERMEDIA_S3_SECRET_ACCESS_KEY")
+USERMEDIA_S3_SIGNING_ENDPOINT_URL = env.str("USERMEDIA_S3_SIGNING_ENDPOINT_URL")
```

Comment:
I'm not sure what are you saying here, can you elaborate a bit further?

Defaults also don't make sense here because we can't possibly expect to know a set of defaults that this will work with. The local docker-compose does have defaults however.

Reply:
It looks like the code does actually already use this, but the env var name implies you would pass something like `s3.<region>.amazonaws.com/sign-url` (which doesn't actually exist as pre-signed URLs are generated client side). `USERMEDIA_S3_ENDPOINT_URL` might be more clear as there isn't an implication that the endpoint is specifically used for generating a signature.

Reply:
So the thing is, the endpoint URL has to be different depending on if we're generating pre-signed URLs vs. if we're doing anything else.

If I remember correctly, the endpoint used for signing has to be the _actual URL_ of the s3 bucket, bypassing all CDNs or whatnot, whereas the regular endpoint could be pointing to the CDN domain (in our case `cdn.thunderstore.io`).

In local development this is also a bit different, since in docker-compose we have
```yml
    USERMEDIA_S3_ENDPOINT_URL: "http://minio:9000/"
    USERMEDIA_S3_SIGNING_ENDPOINT_URL: "http://localhost:9000/"
```

Note that the `USERMEDIA_S3_ENDPOINT_URL` isn't even accessible outside of the docker container, whereas the `USERMEDIA_S3_SIGNING_ENDPOINT_URL` is. However, the `USERMEDIA_S3_SIGNING_ENDPOINT_URL` would not work inside the docker container, thus why this separation is required even in local development already.

Reply:
Since the setting is likely going to be only used once (where the signed URLs are generated), a comment should be fine. I was thinking about using something like `INTERNAL` but as this is the only use it should be fine as long as it's clear what the setting actually is when reading `settings.py`.